code/extensions/Lumberjack.php

```diff
@@ -16,14 +16,15 @@ class Lumberjack extends SiteTreeExtension
     /**
      * Loops through subclasses of the owner (intended to be SiteTree) and checks if they've been hidden.
      *
+     * @var bool $allowSiteTreeVisibility       whether to allow both gridfield and site tree visibility
      * @return array
      **/
-    public function getExcludedSiteTreeClassNames()
+    public function getExcludedSiteTreeClassNames($allowSiteTreeVisibility = false)
     {
         $classes = array();
         $siteTreeClasses = $this->owner->allowedChildren();
         foreach ($siteTreeClasses as $class) {
-            if (Config::inst()->get($class, 'show_in_sitetree') === false) {
+            if ((Config::inst()->get($class, 'show_in_sitetree') === false) || ($allowSiteTreeVisibility && Config::inst()->get($class, 'show_in_lumberjack'))) {
```

Comment:
It's different config values - show_in_sitetree and show_in_lumberjack.

The latter one is only applicable when the function is called to provide a list of pages for the gridfield. Might consider renaming the variable to $showPageInBothSitetreeAndGridfield or similar, but it's not user facing method, it's just internal function.